btp-document/BMR-Deployment.md

```diff
@@ -0,0 +1,212 @@
+
+- Relay 1: listen/collect event's data from Network A --> build `Relay Message` --> push it to ***BMC*** of Network B
+- Relay 2: listen/collect event's data from Network B --> build `Relay Message` --> push it to ***BMC*** of Network A
+
+Before moving on guidance to deploy BMRs, we would like to make short explanation so that the following instructions would be easier to follow up. As mentioned earlier, this example demonstates to connect ICON Network to Moonriver Network
+
+- Confusion point 1: BMR handles a connection from A to B. BMR listens on A, but it must be registered on B
+
+BMR1: listen blocks generating from ICON Network --> build `Relay Message` --> push this message to Moonriver-BMC contract.
+
+==> In order to push `Relay Message` to Moonriver-BMC contract, BMR1 must be registered into Moonriver-BMC contract. Thus, it requires BMR1's address being generated in Moonriver Network. Then, `Operator` of Moonriver-BMC contract can manually add this address to handle a connection link (ICON --> Moonriver)
+
+BMR2: listen blocks generating from Moonriver Network --> build `Relay Message` --> push this message to ICON-BMC contract.
+
+==> In order to push `Relay Message` to ICON-BMC contract, BMR2 must be registered into ICON-BMC contract. Thus, it requires BMR2's address being generated in ICON Network. Then, `Operator` of Moonriver-BMC contract can manually add this address to handle a connection link (Moonriver --> ICON)
+
+- Confusion point 2: BMR handles a connection from A to B. Similar to the above confusion point, BMR listens block generating on A, but it must query this offet through calling `getBMCLinkStatus` on B
+
+==> To avoid chicken-and-egg problem, `Operator` will choose an `offset` and set it into BMV contract when this contract is deployed. From that point, blocks can be verified and synced. Upon successful verification, BMV updates this `offset` to the latest block height. In addition, current block height on A might be higher than current block height was verified by BMV contract on B. Thus, BMR is required to retrieve the latest `offset` from BMV contract through calling `getBMCLinkStatus` on B
+
+### 1. Build executable files
+
+____
+
+```shell
+# INode is a folder that was created when deploying a local node in ICON Network
+$ CONFIG_DIR=path/to/config/folder/INode
+
+$ cd $CONFIG_DIR
+
+#  Clone BTP Project
+$ git clone https://github.com/icon-project/btp.git
+
+$ cd btp && git checkout refactor-layout
+
+# Output binaries are placed under bin/ directory.
+$ make btpsimple
+```
+
+### 2. Generate keystore and configuration files
+
+____
+
+#### Download CLI tools
+
+```shell
+# Get goloop-cli
+$ go get github.com/icon-project/goloop/cmd/goloop
+
+# Get ethkey-cli
+$ go get github.com/ethereum/go-ethereum/cmd/ethkey
+```
+
+#### Create keystore files
+
+- Generate keystore of BMR from Moonriver --> ICON
+
+```shell
+$ goloop ks gen --out $CONFIG_DIR/icon-bmr.keystore.json --password $YOUR_PASSWORD
```

Comment:
Unclear at which point in this part of the tutorial we should command in goloop container or in our local machine?

In our local machine, goloop binary is not available (needs to build it and the path is different)